source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -252,23 +254,23 @@ void AuthenticatorImpl::verifyKey() {
 
 void AuthenticatorImpl::doneWithStatus(const Status& status) {
   // if on allow missing or failed this should verify all tokens, otherwise stop on ok.
-  if ((Status::Ok == status && !is_allow_failed_) || tokens_.empty()) {
+  if ((Status::Ok == status && provider_) || tokens_.empty()) {
```

Comment:
For code readability,  prefer to have a separate variable  allow_missing_or_fail_